### ISSUE TITLE
GNOME 3 crashes on startup when AppArmor is enabled

### ISSUE THREAD

**Issue by emilazy** [NixOS/nixpkgs#77968]
**Describe the bug**
I was futzing for a while trying to figure out why GNOME crashed on startup until I realized that disabling AppArmor fixed it.

**To reproduce**
1. Turn on the `hardened.nix` profile.
2. Turn on `services.xserver.desktopManager.gnome3.enable`.
3. `$ XDG_SESSION_TYPE=wayland dbus-run-session gnome-session`

**Expected behaviour**
GNOME doesn't crash on startup, even with hardening features enabled.

**Metadata**
```
 - host os: `Linux 5.4.12-hardened, NixOS, 20.03.git.0436ef7 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - nixpkgs: `/home/emily/etc/nixos/extern/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

(not sure how to fill this part out, sorry)

**Comment by hedning** [comment]
Hmm, this might be the soft real-time security wrapper. Does it work if you comment this out:

https://github.com/NixOS/nixpkgs/blob/07b8ab4ff791dc8bcfeea643fd9a1fbedc534bbd/nixos/modules/services/x11/desktop-managers/gnome3.nix#L283-L299

**Comment by emilazy** [comment]
Ugh, I'm sorry -- upon trying to reproduce everything mysteriously works; probably some other config change was interfering. Closing this as a false report.

That said, interestingly, the real-time wrapper doesn't seem to actually work:
```
Jan 18 12:50:58 renko .gnome-shell-wr[8301]: Failed to obtain high priority context
Jan 18 12:50:58 renko .gnome-shell-wr[8301]: Failed to obtain high priority context
Jan 18 12:50:58 renko .gnome-shell-wr[8301]: Failed to set RT scheduler: Operation not permitted
```

I don't know whether that's because it's not using the user `gnome-shell-wayland.service` due to me starting the session manually (I get errors about falling back to non-systemd-based startup because of dbus not finding `org.freedesktop.systemd1` when starting GNOME, for reasons I haven't yet been able to figure out), or whether the approach is just broken, but I figured I should note the observed behaviour at least.
